packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -30,32 +22,55 @@ class Carousel extends Foundation<
 > {
     public static displayName: string = `${DisplayNamePrefix}Carousel`;
 
+    public static defaultProps: Partial<CarouselProps> = {
+        autoplay: false,
+    };
+
     /**
      * React life-cycle method
      */
     public static getDerivedStateFromProps(
         nextProps: CarouselProps,
         prevState: CarouselState
-    ): null | CarouselState {
+    ): null | Partial<CarouselState> {
+        const updatedState: Partial<CarouselState> = {};
+
         if (nextProps.activeId && nextProps.activeId !== prevState.activeId) {
-            return {
-                activeId: nextProps.activeId,
-            };
+            updatedState.activeId = nextProps.activeId;
         }
 
-        return null;
+        return !isEmpty(updatedState) ? updatedState : null;
     }
 
     /**
      * Handled props
      */
     protected handledProps: HandledProps<CarouselHandledProps> = {
+        autoplay: void 0,
+        autoplayInterval: void 0,
         managedClasses: void 0,
         label: void 0,
         activeId: void 0,
         items: void 0,
+        onCarouselMouseEnterOrFocus: void 0,
+        onCarouselMouseLeave: void 0,
     };
 
+    /**
+     * Store a reference the autoplay interval
+     */
+    private autoplayInterval: number | null;
```

Comment:
Why are we storing this as a private variable? Could we not just set the 6000 you have as default props?

Reply:
undone and passed as props. Previously I was working to try and handle stopping on focus/mouseover and decided to make the thing entirely controlled, so this can simply be prop driven.